src/OSPSuite.R/Services/DimensionTask.cs

```diff
@@ -92,7 +94,11 @@ public DimensionTask(IDimensionFactory dimensionFactory)
          _dimensionFactory = dimensionFactory;
       }
 
-      public bool HasUnit(string dimensionName, string unit) => DimensionByName(dimensionName).HasUnit(unit);
+      public bool HasUnit(string dimensionName, string unit)
+      {
+         var dimension = DimensionByName(dimensionName);
+         return dimension.HasUnit(unit) || dimension.SupportsUnit(unit, ignoreCase: true);
```

Comment:
for R, I am enhancing this check to return also if there is a case sensitive counter part